tests/unit/amo/components/TestAddon.js

```diff
@@ -748,6 +771,242 @@ describe('Addon', () => {
+
+      sinon.assert.notCalled(fakeDispatch);
+    });
+
+    it('is hidden when an add-on has not loaded yet', () => {
+      const root = shallowRender({ addon: undefined });
+      expect(root.find('.AddonDescription-more-addons'))
+        .toHaveLength(0);
+    });
+
+    it('is hidden when other addons are not loaded yet', () => {
+      const root = shallowRender({ addonsByAuthors: undefined });
+      expect(root.find('.AddonDescription-more-addons'))
+        .toHaveLength(0);
+    });
+
+    it('is hidden when there is no other addon', () => {
+      const root = shallowRender({ addonsByAuthors: [] });
+      expect(root.find('.AddonDescription-more-addons'))
+        .toHaveLength(0);
+    });
+
+    it('displays the developer name when add-on is an extension', () => {
+      const root = mountMoreAddonsByAuthors({
+        addonsByAuthors: [fakeAddon],
+      });
+      expect(root.find('.Card-header'))
+        .toIncludeText('More extensions by Krupa');
+    });
+
+    it('displays the artist name when add-on is a theme', () => {
+      const root = mountMoreAddonsByAuthors({
+        addon: fakeTheme,
+        addonsByAuthors: [fakeTheme],
+      });
+      expect(root.find('.Card-header'))
+        .toIncludeText('More themes by Madonna');
+    });
+
+    it('displays the author name when add-on is not extension or theme', () => {
+      const root = mountMoreAddonsByAuthors({
+        addon: {
+          ...fakeAddon,
+          type: ADDON_TYPE_OPENSEARCH,
+        },
+        addonsByAuthors: [fakeAddon],
+      });
+      expect(root.find('.Card-header'))
+        .toIncludeText('More add-ons by Krupa');
+    });
+
+    it('displays a plural form when extension has multiple authors', () => {
+      const root = mountMoreAddonsByAuthors({
+        addon: {
+          ...fakeAddon,
+          authors: Array(2).fill(fakeAddon.authors[0]),
+        },
+        addonsByAuthors: [fakeAddon],
+      });
+      expect(root.find('.Card-header'))
+        .toIncludeText('More extensions by these authors');
+    });
+
+    it('displays a plural form when theme has multiple authors', () => {
+      const root = mountMoreAddonsByAuthors({
+        addon: {
+          ...fakeTheme,
+          authors: Array(2).fill(fakeTheme.authors[0]),
+        },
+        addonsByAuthors: [fakeTheme],
+      });
+      expect(root.find('.Card-header'))
+        .toIncludeText('More themes by these authors');
+    });
+
+    it('displays a plural form when add-on has multiple authors', () => {
+      const root = mountMoreAddonsByAuthors({
+        addon: {
+          ...fakeAddon,
+          authors: Array(2).fill(fakeAddon.authors[0]),
+          type: ADDON_TYPE_OPENSEARCH,
+        },
+        addonsByAuthors: [fakeAddon],
+      });
+      expect(root.find('.Card-header'))
+        .toIncludeText('More add-ons by these authors');
+    });
+
+    it('displays more add-ons by authors', () => {
+      const addonsByAuthors = [
```

Comment:
Unless you can convince me otherwise, I think all mapped props should be set with a dispatch, not raw state. This helps keep everything in sync.

```js
store.dispatch(loadOtherAddonsByAuthors({ 
  slug: fakeAddon.slug, 
  addons: [
    { ...fakeAddon, slug: 'addon-1' },
    { ...fakeAddon, slug: 'addon-2' },
    { ...fakeAddon, slug: 'addon-3' },
  ], 
}));
const root = mountMoreAddonsByAuthors({ store });
```

We usually create a helper specific to the test suite like `_loadOtherAddonsByAuthors()`.